File: Client/digitalengineers.client/src/types/project.ts
import type { ProfessionType, LicenseRequirement, LicenseType, Profession } from './lookup';

// Re-export LicenseType for backward compatibility
export type { LicenseType } from './lookup';

// Project Status Enum
export enum ProjectStatus {
	QuotePending = 'QuotePending',
	Draft = 'Draft',
	QuoteSubmitted = 'QuoteSubmitted',
	QuoteAccepted = 'QuoteAccepted',
	QuoteRejected = 'QuoteRejected',
	InitialPaymentPending = 'InitialPaymentPending',
	InitialPaymentComplete = 'InitialPaymentComplete',
	InProgress = 'InProgress',
	Completed = 'Completed',
	Cancelled = 'Cancelled'
}

// Project Scope Enum
export enum ProjectScope {
	Small = 1,
	Medium = 2,
	Large = 3
}

// Project Management Type Enum
export enum ProjectManagementType {
	ClientManaged = 'ClientManaged',
	DigitalEngineersManaged = 'DigitalEngineersManaged'
}

export interface ProjectFormData {
	// Step 1
	name: string;
	professionTypeIds: number[]; // NEW: Replace licenseTypeIds
	
	// Deprecated - keep for backward compatibility
	licenseTypeIds?: number[];

	// Step 2
	streetAddress: string;
	city: string;
	state: string;
	zipCode: string;
	projectScope: 1 | 2 | 3;
	managementType: ProjectManagementType;

	// Step 3
	description: string;
	files: File[];
	thumbnail: File | null;
}

export interface CreateProjectRequest {
	name: string;
	professionTypeIds: number[]; // NEW
	licenseTypeIds?: number[]; // Deprecated
	streetAddress: string;
	city: string;
	state: string;
	zipCode: string;
	projectScope: number;
	managementType: ProjectManagementType;
	description: string;
	files: File[];
	thumbnail: File | null;
}

// Synchronized with ProjectViewModel from server
export interface ProjectDto {
	id: number;
	name: string;
	description: string;
	status: string;
	clientId?: string;
	clientName?: string;
	clientProfilePictureUrl?: string;
	createdAt: string;
	thumbnailUrl?: string;
	streetAddress: string;
	city: string;
	state: string;
	zipCode: string;
	projectScope: number;
	managementType: string;
	licenseTypeIds: number[];
	quotedAmount?: number;
	taskCount: number;
}

// Synchronized with ProjectDetailsViewModel from server
export interface ProjectDetailsDto {
	id: number;
	name: string;
	description: string;
	status: string;
	clientId: string;
	clientName: string;
	clientEmail: string;
	clientProfilePictureUrl?: string;
	streetAddress: string;
	city: string;
	state: string;
	zipCode: string;
	projectScope: number;
	managementType: string;
	licenseTypeIds: number[];
	licenseTypes: LicenseType[];
	professionTypeIds: number[];
	professions: Profession[];
	createdAt: string;
	updatedAt: string;
	thumbnailUrl?: string;
	files: ProjectFile[];
	taskCount: number; // Add taskCount to match ProjectDto
	// Quote fields
	quotedAmount?: number;
	quoteSubmittedAt?: string;
	quoteAcceptedAt?: string;
	quoteRejectedAt?: string;
	quoteNotes?: string;
}

export interface ProjectFile {
	id: number;
	fileName: string;
	fileUrl: string;
	fileSize: number;
	contentType: string;
	uploadedAt: string;
}

export interface SpecialistLicenseInfo {
	licenseTypeId: number;
	licenseTypeName: string;
	professionId: number;
	professionName: string;
}

// Synchronized with ProjectTeamMemberDto from server
// Universal interface for both assigned specialists and pending bids
export interface ProjectSpecialistDto {
	specialistId: number;
	userId: string;
	name: string;
	profilePictureUrl?: string;
	role?: string; // Role/Title from ProjectSpecialist
	isAssigned: boolean; // true = assigned, false = pending bid
	isAnonymized: boolean; // true = hide real data (DE managed), false = show real data
	assignedOrBidSentAt: string;
	licenseTypes: SpecialistLicenseInfo[];
}

// NEW: For displaying selected profession types
export interface SelectedProfessionType {
	professionType: ProfessionType;
	licenseRequirements: LicenseRequirement[];
}

File: Server/DigitalEngineers.Infrastructure/Entities/Project.cs
using DigitalEngineers.Domain.Enums;

namespace DigitalEngineers.Infrastructure.Entities;

public class Project
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public ProjectStatus Status { get; set; } = ProjectStatus.Draft;
    
    // Client relationship
    public string ClientId { get; set; } = string.Empty;
    
    // Location
    public string StreetAddress { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string State { get; set; } = string.Empty;
    public string ZipCode { get; set; } = string.Empty;
    
    // Project details
    public ProjectScope ProjectScope { get; set; }
    public ProjectManagementType ManagementType { get; set; } = ProjectManagementType.DigitalEngineersManaged;
    public string? ThumbnailUrl { get; set; }
    public decimal Budget { get; set; }
    public DateTime? StartDate { get; set; }
    
    // Quote fields
    public decimal? QuotedAmount { get; set; }
    public DateTime? QuoteSubmittedAt { get; set; }
    public DateTime? QuoteAcceptedAt { get; set; }
    public DateTime? QuoteRejectedAt { get; set; }
    public string? QuoteNotes { get; set; }
    
    // Timestamps
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    
    // Relationships
    public ICollection<ProjectLicenseType> ProjectLicenseTypes { get; set; } = [];
    public ICollection<ProjectFile> Files { get; set; } = [];
    public ICollection<ProjectSpecialist> AssignedSpecialists { get; set; } = [];
    public ICollection<BidRequest> BidRequests { get; set; } = [];
    public ICollection<Review> Reviews { get; set; } = [];
}
